force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -81,15 +84,17 @@ public with sharing class ClassesObjectsMethods {
      */
     public static String removeDuplicateWords(String sentence) {
         // Split the sentence into words
-
+        List<String> words = sentence.split(' ');
         // Create a set to store the unique words
-
+        Set<String> stringSet = new Set<String>();
         // Add each word to the set
-
+        for (String word : words) {
+            stringSet.add(word);
+        }
         // Join the unique words back into a sentence
-
+        String finalSentence = String.join(stringSet, ' ');
```

Comment:
Try to add the set to a new list in order to add it using the String.join() method